aw_client/client.py

```diff
@@ -137,18 +143,30 @@ def insert_events(self, bucket_id: str, events: List[Event]) -> None:
         data = [event.to_json_dict() for event in events]
         self._post(endpoint, data)
 
-    def heartbeat(self, bucket_id: str, event: Event, pulsetime: float, queued: bool=False) -> Optional[Event]:
+    def heartbeat(self, bucket_id: str, event: Event, pulsetime: float, queued: bool=False):
         """ This endpoint can use the failed requests retry queue.
             This makes the request itself non-blocking and therefore
             the function will in that case always returns None. """
-
-        endpoint = "buckets/{}/heartbeat?pulsetime={}".format(bucket_id, pulsetime)
-        data = event.to_json_dict()
-        if queued:
-            self.request_queue.add_request(endpoint, data)
-            return None
+        commit = False
```

Comment:
Nice work, but two things:

 1. The entire pre-merging thing should only be run if the call has `queued` set to `True`.
     - That is, you can wrap the simple no-merge logic in a `if not queued:` or equivalent before all the queued/merge-specific code.
 2. The event-merging here should use [heartbeat_merge](https://github.com/ActivityWatch/aw-core/blob/master/aw_core/transforms.py#L45) or [heartbeat_reduce](https://github.com/ActivityWatch/aw-core/blob/master/aw_core/transforms.py#L29) instead.
     - Will make the logic simpler.
     - Once events are merged, simply set `pulsetime = (now - event.timestamp) + pulsetime`
     - An option to doing this in the `heartbeat` function merging *in the queue* (using `heartbeat_reduce`), but not sure if we dare poke around in the queue code.

Reply:
I've been trying to fix this now for a couple of hours without success. There are a lot of corner cases not working properly with heartbeat_merge for example when using durations and heartbeat timeperiods which are smaller than last_event.timestamp+last_event.duration. Will continue trying to fix it but it is not as trivial as it seemed from the start.

Also I realized that it's a bad idea to use `now` in this library since someone might want to send a heartbeat which is in the past or in the future.

Reply:
The whole thing with durations on heartbeats is completely broken currently with merged_events as well as the solution i proposed here and is rather hard to fix. This makes this client-side caching work well with aw-watcher-window but is broken with aw-watcher-afk since it utilizes heartbeats with durations.

Reply:
Could you provide some examples of events that wouldn't work with the current heartbeat_merge implementation? 

I'm thinking of writing some unit tests and give it a shot.